packages/db/fauna/resources/Function/createBatch.js

```diff
@@ -0,0 +1,81 @@
+import fauna from 'faunadb'
+
+const {
+  Function,
+  CreateFunction,
+  Query,
+  Lambda,
+  Let,
+  Select,
+  Var,
+  If,
+  Create,
+  Update,
+  Exists,
+  Collection,
+  Foreach,
+  Index,
+  IsEmpty,
+  Abort,
+  Match,
+  Do,
+  Get
+} = fauna
+
+const name = 'createBatch'
+const body = Query(
+  Lambda(
+    ['data'],
+    Let(
+      {
+        batchCid: Select('batchCid', Var('data')),
+        pieceCid: Select('pieceCid', Var('data')),
+        batch: Create(
+          Collection('Batch'),
+          {
+            data: {
+              cid: Var('batchCid'),
+              pieceCid: Var('pieceCid')
+            }
+          }
+        ),
+        batchRef: Select('ref', Var('batch')),
+        entries: Select('entries', Var('data'))
```

Comment:
You can/should make all of these mandatory fields to catch errors on my side.

Reply:
`Select` will throw if the property doesn't exist in the object and no default is given (like here).

Reply:
<img width="230" alt="Screenshot 2021-07-12 at 14 33 02" src="https://user-images.githubusercontent.com/152863/125296329-16719a80-e31e-11eb-8e6d-50721af15d45.png">

They are also enforced required by GraphQL (the `!`)

Reply:
Ah got it, all good!